### ISSUE TITLE
Huge memory usage when opening geojson file

### ISSUE THREAD

**Issue by jorisvandenbossche** [Toblerity/Fiona#624]
I experience a memory issue when reading a relatively big GeoJSON file (using `geopandas.read_file`, but I tracked it down to simply opening the file with fiona, so not even iterating through the features).

It is for a very specific file (the GeoJSON file downloaded from here: https://opendata.paris.fr/explore/dataset/les-arbres/export/), which is 104 MB on disk, after reading it in with geopandas it indicates that the dataframe has a memory usage of 150 MB (which will not count the GEOS objects, but since it are Points, I suspect this will not much more). But while reading it in, memory usage increases with 1.7 GB, and also after reading and after deleting the dataframe, the memory usage stays that high.

Script to reproduce (file: https://opendata.paris.fr/explore/dataset/les-arbres/download/?format=geojson&timezone=Europe/Berlin):

```python
import psutil
import gc

p = psutil.Process()
print("At the start: ", p.memory_info().rss / 1e6)

import fiona
print(fiona.__version__)
print("After importing fiona: ", p.memory_info().rss / 1e6)

path = "./les-arbres.geojson"

c = fiona.collection(path)
print("After fiona open ", p.memory_info().rss / 1e6)

del c
print("After del collection: ", p.memory_info().rss / 1e6)

gc.collect()
print("After gc collect: ", p.memory_info().rss / 1e6)
```

which gives:

```
At the start:  12.660736
1.7.11
After importing fiona:  46.239744
After fiona open  1548.541952
After del collection:  1548.541952
After gc collect:  1442.689024
```

(the numbers are ~MBs, and here it is a bit lower than the 1.7 GB I mentioned above, because I am not actually reading the full file).

It is certainly a quite big GeoJSON file, but there is still clearly going something wrong I think? (1.7GB memory usage of 104 MB file)

---

I am using fiona 1.7.11 (installed from conda-forge, with GDAL 2.2.4, on Python 3.5, Ubuntu 16.04). I have seen the IO implementation has changed a bit in master (the MemoryFile), but I did not yet have the time to get master installed to test.

**Comment by jorisvandenbossche** [comment]
If you have pointers on what I can further do to help track this down, all ears. 
Just to note, adding a `c.close()` (or using the `with ..` context that does that automatically) doesn't change anything.

**Comment by sgillies** [comment]
@jorisvandenbossche I think this may be a manifestation of https://trac.osgeo.org/gdal/ticket/6540. Upgrading to GDAL 2.3.0 should help.

**Comment by jorisvandenbossche** [comment]
I tested with GDAL 2.3.1 from conda and fiona master, and I can confirm this is solved then. Only opening the file does not increase the memory at all now.

**Comment by jorisvandenbossche** [comment]
However, actually reading the file (iterating through the collection, or with geopandas), now prints this for every row of the geojson file:

```
...
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
Skipping field geo_point_2d: invalid type 3
...
```

Something unrelated to this issue of course, but is this a known issue or do I open a new issue?

**Comment by jorisvandenbossche** [comment]
Opened https://github.com/Toblerity/Fiona/issues/627 for that.

Another question: is Fiona 1.7.13 already compatible with GDAL 2.3? Or is that support only added for Fiona 1.8? (I looked in the changelog, but didn't see a mention about when it was added)

**Comment by sgillies** [comment]
Will be fixed in 1.8.0.
